File: Front/src/app/Services/CommentaireService/commentaire.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from "@angular/common/http";
import {catchError, Observable, throwError} from "rxjs";
import {Commentaire} from "../../Modules/CommentaireModule/Commentaire.Module";

@Injectable({
  providedIn: 'root'
})
export class CommentaireService {
  private apiURL = 'http://localhost:9090/api/services/commentaire'; // Adjust the port number if needed

  constructor(private http: HttpClient) { }

  getCommentaireById(idCommentaire: number): Observable<Commentaire> {
    return this.http.get<Commentaire>(`${this.apiURL}/getcmtparid/${idCommentaire}`);
  }

  getAllCommentaires(): Observable<Commentaire[]> {
    return this.http.get<Commentaire[]>(`${this.apiURL}/getallCommentaire`);
  }

  updateCommentaire(commentaire: Commentaire): Observable<Commentaire> {
    return this.http.put<Commentaire>(`${this.apiURL}/updateCommentaire`, commentaire);
  }

  deleteCommentaire(idCommentaire: number): Observable<boolean> {
    return this.http.delete<boolean>(`${this.apiURL}/deletecmt/${idCommentaire}`);
  }

  getCommentsByPostId(idPost: number): Observable<Commentaire[]> {
    return this.http.get<Commentaire[]>(`${this.apiURL}/getcmtparpost/${idPost}`);
  }

  addCommentToPost(commentaire: Commentaire, idPost: number): Observable<Commentaire> {
    return this.http.post<Commentaire>(`${this.apiURL}/addCommentToPost/${idPost}`, commentaire)
      .pipe(
        catchError(error => {
          // Handle the error here
          console.error('Error creating comment', error);
          // Throw an observable with a user-facing error message
          return throwError(error);
        })
      );
  }


}

File: Back/src/main/java/com/example/back/Services/CommentaireService.java
package com.example.back.Services;

import com.example.back.Entities.Commentaire;

import java.util.List;

public interface CommentaireService {
    Commentaire createCommentaire(Commentaire commentaire);
    Commentaire getCommentaireById(Long idCommentaire);
    List<Commentaire> getAllCommentaires();
    Commentaire updateCommentaire( Commentaire commentaire);
    Boolean deleteCommentaire(Long idCommentaire);
    List<Commentaire> getCommentsByPostId(Long idPost);
    Commentaire createCommentaireWithPost(Commentaire commentaire, Long postId);
    Commentaire addCommentToPostAndAssignToUser(Commentaire commentaire, Long postId, Long userId);


}
